dali/operators/decoder/inflate/inflate_params.h

```diff
@@ -0,0 +1,397 @@
+      InferOffsetsFromSizes(in_shape);
+    }
+  }
+
+  template <typename T, typename InShape, typename ChunkValidator>
+  void ValidateChunks(const TensorListView<StorageCPU, const T, 1> &samples,
+                      const InShape &in_shape, ChunkValidator &&chunk_validator) {
+    for (int sample_idx = 0; sample_idx < in_shape.num_samples(); sample_idx++) {
+      auto sample_num_bytes = in_shape[sample_idx].num_elements();
+      const auto *chunks_data = samples.data[sample_idx];
+      for (int chunk_idx = 0; chunk_idx < samples.shape[sample_idx].num_elements(); chunk_idx++) {
+        chunk_validator(sample_idx, sample_num_bytes, chunks_data[chunk_idx]);
+      }
+    }
+  }
+
+  template <typename InOffsetT, typename InSizeT, typename TLShape>
+  void ValidateSizesAgainstOffsets(const TensorListView<StorageCPU, const InOffsetT, 1> &offsets,
+                                   const TensorListView<StorageCPU, const InSizeT, 1> &sizes,
+                                   const TLShape &in_shape) {
+    auto batch_size = in_shape.num_samples();
+    for (int sample_idx = 0; sample_idx < batch_size; sample_idx++) {
+      auto num_chunks = offsets.shape[sample_idx].num_elements();
+      DALI_ENFORCE(
+          num_chunks == sizes.shape[sample_idx].num_elements(),
+          make_string("The number of elements passed as `", offsetArgName, "` and `", sizeArgName,
+                      "` must match for corresponding samples. However for sample of idx ",
+                      sample_idx, " there are ", num_chunks, " offsets and ",
+                      sizes.shape[sample_idx].num_elements(), " sizes."));
+    }
+    for (int sample_idx = 0; sample_idx < batch_size; sample_idx++) {
+      auto num_chunks = offsets.shape[sample_idx].num_elements();
+      auto sample_num_bytes = in_shape[sample_idx].num_elements();
+      const auto *chunk_offsets = offsets.data[sample_idx];
+      const auto *chunk_sizes = sizes.data[sample_idx];
+      for (int chunk_idx = 0; chunk_idx < num_chunks; chunk_idx++) {
+        int64_t chunk_end = chunk_offsets[chunk_idx];
+        chunk_end += chunk_sizes[chunk_idx];
+        DALI_ENFORCE(
+            chunk_end <= sample_num_bytes,
+            make_string("Input chunk cannot exceed the sample size. However, for a sample of idx ",
+                        sample_idx, ", got a chunk that starts at the position ",
+                        chunk_offsets[chunk_idx], " and has size ", chunk_sizes[chunk_idx],
+                        " while the sample contains only ", sample_num_bytes, " bytes."));
+      }
+    }
+  }
+
+  template <typename TLShape>
+  void InferSizesFromOffsets(const TLShape &in_shape) {
+    assert(offsets_.size() == chunks_per_sample_.num_elements());
+    auto last_chunk_error_msg = [](auto sample_idx) {
+      return make_string(
+          "The last chunk's offset exceeds the size of the sample for sample of idx ", sample_idx,
+          ".");
+    };
```

Comment:
This will never happen - offset validation makes it impossible.

Reply:
removed it.